third_party/checker_framework_dataflow/BUILD

```diff
@@ -4,23 +4,16 @@ licenses(["restricted"])  # GNU GPL v2 with Classpath exception
 
 filegroup(
     name = "srcs",
-    srcs = glob(["**"]),
+    srcs = ["dataflow-2.2.2-sources.jar"],
```

Comment:
These sources get collected into a tar archive by `//:bazel-srcs`. Currently the tar contains the `.java` sources, this changes causes the tar to contain the source jar. I think we want the preserve the current behaviour.

Reply:
This PR is a principle change in handling of checker framework. We stop building from the source and start to consume the pre-built artifacts (downloaded from Maven Central).

If we do not build from the source, and do not have sources in our tree, why should we supply sources in `:bazel-srcs` at all? We don't do this neither for `guava`, nor for `error prone` and nor for custom java compiler.

In fact, before this change, the `//:bazel-srcs` tar already contains some source JARs:

```
  $ davido@wizball:~/projects/bazel (master %=)$ tar -tf bazel-bin/bazel-srcs.tar | grep --regexp\="third_party/.*jar$" | grep source
./third_party/asm/asm-6.0-sources.jar
./third_party/asm/asm-analysis-6.0-sources.jar
./third_party/asm/asm-commons-6.0-sources.jar
./third_party/asm/asm-tree-6.0-sources.jar
./third_party/asm/asm-util-6.0-sources.jar
./third_party/java/jarjar/jarjar-core/src/test/resources/enumtest.jar
```

After this change we have two more:

```
[...]
./third_party/checker_framework_dataflow/dataflow-2.2.2-sources.jar
./third_party/checker_framework_javacutil/javacutil-2.2.2-sources.jar
```

So, if you would argue, that we should stop include source JARs for checker framework in `//:bazel-srcs` artifact I think would agree with this point.

A different point is third-party:gpl-srcs. There we also include source JARs for the checker-framework that is part of embeded zip. But there we also have JARs, Moreover, I also think, that a follow-up change could be to stop building `jFormatString` from the source (another LGPL licensed library) and consume pre-built binary from the Central:

```
<dependency>
    <groupId>com.google.code.findbugs</groupId>
    <artifactId>jFormatString</artifactId>
    <version>3.0.0</version>
</dependency>

```

Reply:
> Should we rather remove checker framework source JARs from `//:bazel-srcs` entirely?

I discontinued packaging checker framework in `//:bazel-srcs` in the updated PR, and extended the commit message:

```
This change stops building checker framework from the source and consumes
the artifacts from Central. As the consequence, the sources are removed
from the Bazel tree and we discontinue packaging the sources in Bazel
source artifact.

LGPL sources are still have to be included in embeded_tools archive. The
source jars are included in embedded_tools archive now:

$ unzip -t bazel-genfiles/src/embedded_tools.zip|grep checker_framework
testing: third_party/checker_framework_dataflow/dataflow-2.2.2-sources.jar
testing: third_party/checker_framework_javacutil/javacutil-2.2.2-sources.jar
```

Reply:
> Moreover, I also think, that a follow-up change could be to stop building jFormatString from the source (another LGPL licensed library) and consume pre-built binary from the Central ...

Done in https://github.com/bazelbuild/bazel/pull/4843